File: php/src/ApiInternal/EndpointAutoConfig.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\ApiInternal;

use Svix\Models\EndpointOut;
use Svix\Models\SubscribeIn;
use Svix\Request\SvixHttpClient;

class EndpointAutoConfig
{
    public function __construct(
        private readonly SvixHttpClient $client,
    ) {
    }

    /** Update an auto-config endpoint by providing endpoint details. */
    public function update(
        string $appId,
        string $endpointId,
        SubscribeIn $subscribeIn,
    ): EndpointOut {
        $request = $this->client->newReq('PUT', "/api/v1/app/{$appId}/endpoint/{$endpointId}/auto-config");
        $request->setBody(json_encode($subscribeIn));
        $res = $this->client->send($request);

        return EndpointOut::fromJson($res);
    }
}

File: kotlin/lib/src/main/kotlin/internalapi/EndpointAutoConfig.kt
// this file is @generated
package com.svix.kotlin.internal

import com.svix.kotlin.SvixHttpClient
import com.svix.kotlin.models.EndpointOut
import com.svix.kotlin.models.SubscribeIn

class EndpointAutoConfig(private val client: SvixHttpClient) {
    /** Update an auto-config endpoint by providing endpoint details. */
    suspend fun update(appId: String, endpointId: String, subscribeIn: SubscribeIn): EndpointOut {
        val url =
            client
                .newUrlBuilder()
                .encodedPath("/api/v1/app/$appId/endpoint/$endpointId/auto-config")

        return client.executeRequest<SubscribeIn, EndpointOut>(
            "PUT",
            url.build(),
            reqBody = subscribeIn,
        )
    }
}
